qutip/wigner.py

```diff
@@ -581,16 +594,33 @@ def qfunc(state, xvec, yvec, g=sqrt(2)):
 
     qmat = zeros(size(amat))
 
+    if precompute is None:
+        # Decide whether precomputation should be used, if not already set
+        precompute = isoper(state)
+
+    memory = len(xvec) * len(yvec) * state.shape[0] * 16 / 1024**2
+    if (precompute is True) and (memory > 1024):
```

Comment:
It would be nice if there was a way for a user to ask for precompute to happen even if memory usage is `> 1024`. My suggestion is to have precompute never happen if `precompute=False` is passed, always happen if `precompute=True`, and happen if `isoper(state) and memory > 1024` is `precompute=None`.

Reply:
There is, as I understand it - you just manually have to call `qfunc_precompute` with the increased memory limits, and pass that as the argument to this function.  Now I think of it, though, that means you can't do `if precompute` or `if precompute == True`, which presumably the author realised (since `x is True` isn't a very natural expression).  Still, `precompute is True` _feels_ wrong, and it could perhaps be a little more obvious from within the code that `precompute` is allowed to be both a Boolean and a numpy array (I know the docstring says it).

I'm not super opposed to it being a little difficult and non-intuitive to ask for precomputation of over a gigabyte, including needing you to be very explicit.  The amount of memory stored is an implementation detail, and the size could be rather surprising, so gating it behind an "advanced" function to prevent you from crashing your Python session accidentally doesn't seem terrible (though I agree too that a hard-coded warning value is very much nonideal).